xarray/backends/api.py

```diff
@@ -15,35 +15,53 @@
 from ..core.combine import auto_combine
 from ..core.utils import close_on_error, is_remote_uri
 from ..core.pycompat import basestring, path_type
+from ..core.options import OPTIONS
 
 DATAARRAY_NAME = '__xarray_dataarray_name__'
 DATAARRAY_VARIABLE = '__xarray_dataarray_variable__'
 
 
+def _guess_file_format(path):
+    """Make an educated guess about the file format based on the first few
+    bytes.
+
+    If the file does not exist, assume netcdf3.
+    """
+    try:
+        f = open(path, 'rb')
+    except IOError:
+        return 'netcdf3'
```

Comment:
Is this really better than raising an error for a missing file? I think it's probably fine to surface the default error message from `open()`, which looks something like: `FileNotFoundError: [Errno 2] No such file or directory: 'asdolfkjalsdkfj'`

Reply:
There were tests that needed to save data to a netCDF file and they were failing here because these files did not exist. Recommending `netcdf3` here for such cases made sense to me because it enabled selection of an appropriate I/O engine (either `netcdf4` or `scipy`).